docs/guides/hybrid.md

```diff
@@ -0,0 +1,135 @@
+[remote debugger](https://github.com/appium/appium-remote-debugger).
+This debugger can connect directly to any WebKit debugger socket exposed by the system.
+The protocol used for the communication there is a proprietary Apple's JSON RPC similar
+to Chrome's [Devtools Protocol](https://chromedevtools.github.io/devtools-protocol/).
+Not all web views expose debugger web sockets by default thus making them invisible
+for the XCUITest driver and not showing in the available contexts list.
+Make sure the following prerequisites are satisfied if you are unsure about whether
+the particular web view is debuggable or not:
+
+- If you use real devices then make sure the Settings→Safari→Advanced→Web Inspector
+  checkbox is turned on.
+- If your app's web view is based on WKWebView then make sure the
+  [isInspectable](https://developer.apple.com/documentation/webkit/wkwebview/4111163-inspectable?language=objc) property of it set to `true`. Note, that you must have access to the application sources in order
+  to ensure that!
+- Make sure you see the corresponding web view in Safari's
+  [remote debugger](https://help.salesforce.com/s/articleView?id=000391692&type=1) list.
+
+If all the above requirements have been satisfied, but the desired web view is still not present in the
+XCUITest driver's context list then there is probably an issue in the driver itself, which must be reported
+to driver maintainers.
+
+### Entering the web view context
+
+Here are the steps required to talk to a web view in your XCUITest driver test:
+
+1. Navigate to a portion of your app where a web view is active
+2. Retrieve the currently available contexts
+    * This returns a list of contexts we can access, like `'NATIVE_APP'` or `'WEBVIEW_1'`
+3. Set the id of the context you want to access
+    * This puts your XCUITest session into a mode where all commands are
+      interpreted as being intended for automating the web view, rather than the
+      native portion of the app. For example, if you run `findElement`, it
+      will operate on the DOM of the web view, rather than return native elements.
+      Of course, certain WebDriver methods only make sense in one context or
+      another, so in the wrong context you will receive an error message.
+4. To stop automating in the web view context and go back to automating the
+   native portion of the app, simply set the context
+   again with the native context id (generally `'NATIVE_APP'`) to leave the web
+   context and once again access the native commands.
+
+### Automatically entering the web view context on session start
+
+If your application begins in a web view, and you do not want to automate the
+native application before entering it, you can have XCUITest driver automatically enter
+the web view context on session initialization by setting the `autoWebview`
+[capability](../reference/capabilities.md) to `true`.
+
+
+### Examples
```

Comment:
Can these code examples be presented using tabs for each language, like in https://github.com/appium/appium-xcuitest-driver/blob/master/docs/reference/execute-methods.md ?

Reply:
updated